### PR TITLE
Add coverage for five low-hanging Elixir modules

### PR DESCRIPTION
**Pull request by frantic-openai**
#### Context

MT-715 asks us to remove the lowest-hanging ignored modules from
`elixir/mix.exs` and keep the Elixir app at 100% coverage.

#### TL;DR

*Remove five easy modules from the coverage ignore list and add the
tests needed to keep total coverage at 100%.*

#### Summary

- Remove `Mix.Tasks.Specs.Check`, `SymphonyElixir`,
  `SymphonyElixir.Application`, `SymphonyElixir.Workflow`, and
  `SymphonyElixir.PromptBuilder` from the coverage ignore list.
- Add tests for workflow loading edge cases, the `SymphonyElixir`
  wrapper entrypoint, nested prompt serialization, and the
  `specs.check` mix task.
- Make the dashboard coalescing test deterministic by isolating it
  from the shared orchestrator instead of widening its timeout.

#### Alternatives

- Leave the five modules ignored and only document the gap. That does
  not satisfy MT-715.
- Target larger ignored modules like `Workspace` or `CLI` first. They
  need more setup-heavy coverage for less ticket progress.

#### Test Plan

- [x] `make -C elixir all`
- [x] `cd elixir && mix test test/symphony_elixir/core_test.exs test/mix/tasks/specs_check_task_test.exs`
- [x] `cd elixir && mix test test/symphony_elixir/orchestrator_status_test.exs:1068`

### CODE REVIEW

File: elixir/test/symphony_elixir/orchestrator_status_test.exs [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ defmodule SymphonyElixir.OrchestratorStatusTest do
     end)
 
     StatusDashboard.notify_update(dashboard_name)
-    assert_receive {:render, first_render_ms, _content}, 200
+    assert_receive {:render, first_render_ms, _content}, 500
```
Comment (frantic-openai) [inline elixir/test/symphony_elixir/orchestrator_status_test.exs]: why 500?

Comment (frantic-openai) [reply elixir/test/symphony_elixir/orchestrator_status_test.exs]: Kept the original 200ms assertions. Instead of widening the timeout, this test now temporarily stops the shared `SymphonyElixir.Orchestrator` so `StatusDashboard` takes the fast snapshot path during the coalescing check, then restarts the orchestrator on exit. That keeps the behavior under test intact and `make -C elixir all` passes on commit `c28839d`.

Comment (frantic-openai) [reply elixir/test/symphony_elixir/orchestrator_status_test.exs]: [codex] Addressed in `c28839d`: kept the original 200ms assertions and made the test deterministic by temporarily stopping the shared `SymphonyElixir.Orchestrator` during the coalescing check, then restarting it on exit. Validation rerun: `cd elixir && mix test test/symphony_elixir/orchestrator_status_test.exs:1068` and `make -C elixir all`.
